examples/gno.land/p/demo/microblog/microblog.gno

```diff
@@ -0,0 +1,198 @@
+package microblog
+
+import (
+	"errors"
+	"sort"
+	"std"
+	"strings"
+	"time"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+var (
+	ErrNotFound    error  = errors.New("not found")
+	StatusNotFound string = "404"
+)
+
+type Microblog struct {
+	Title  string
+	Prefix string   // i.e. r/gnoland/blog:
+	Pages  avl.Tree // author (string) -> Page
+}
+
+func NewMicroblog(title string, prefix string) (m *Microblog) {
+	return &Microblog{
+		Title:  title,
+		Prefix: prefix,
+		Pages:  avl.Tree{},
+	}
+}
+
+func (m *Microblog) GetPages() []*Page {
+	var (
+		pages = make([]*Page, m.Pages.Size())
+		index = 0
+	)
+
+	m.Pages.Iterate("", "", func(key string, value interface{}) bool {
+		pages[index] = value.(*Page)
+		index++
+		return false
+	})
+
+	sort.Sort(ByLastPosted(pages))
+
+	return pages
+}
+
+func (m *Microblog) RenderHome() string {
+	output := ufmt.Sprintf("# %s\n\n", m.Title)
+	output += "# pages\n\n"
+
+	for _, page := range m.GetPages() {
+		output += ufmt.Sprintf("- [%s](%s%s)\n", page.Author.String(), m.Prefix, page.Author.String())
+	}
+
+	return output
+}
+
+func (m *Microblog) RenderUser(user string) string {
+	silo, found := m.Pages.Get(user)
+	if !found {
+		return StatusNotFound
+	}
+
+	return (silo.(*Page)).String()
+}
+
+func (m *Microblog) Render(path string) string {
+	parts := strings.Split(path, "/")
+
+	isHome := path == ""
+	isUser := len(parts) == 1
+
+	switch {
+	case isHome:
+		return m.RenderHome()
+
+	case isUser:
+		return m.RenderUser(parts[0])
+	}
+
+	return StatusNotFound
+}
+
+func (m *Microblog) newPage() error {
+	author := std.GetOrigCaller()
+	_, found := m.Pages.Get(author.String())
+	if found {
+		return errors.New("author already exists")
+	}
```

Comment:
Seems a bit redundant if the only user of this function is `NewPost`, which has already established when calling `newPage` that the page doesn't exist yet?

What if we removed this and changed `NewPost` to just call `m.Pages.Set`?

Reply:
Yeah! That makes a lot of sense. Fixed with 69d9d777